plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeSessionProperties.java

```diff
@@ -348,4 +354,9 @@ public static boolean isCheckpointFilteringEnabled(ConnectorSession session)
     {
         return session.getProperty(CHECKPOINT_FILTERING_ENABLED, Boolean.class);
     }
+
+    public static boolean storeTableMetadata(ConnectorSession session)
```

Comment:
isStoreTableMetadataInMetastoreEnabled()

Reply:
btw. do we want this to be a session toggle rather than a config toggle? Doesn't make sense for user to toggle it on and off.

Reply:
This is a kill switch. Made it a hidden session property.